services/controllerhost/api_handlers.go

```diff
@@ -353,6 +357,13 @@ func refreshInputHostsForDst(context *Context, dstUUID string, now int64) ([]str
 	}
 
 	var dstType = getDstType(dstDesc)
+
+	// Fail attempts to publish to Kafka destinations
+	if dstType == dstTypeKafka {
```

Comment:
I don't really like the idea of storing an empty set of ip:ports and returning them -- since we optimizing for a very uncommon case; if at all, I'd rather update the cache to support and remember to return an "error" on the request.

And yes, we should probably reject these in the Frontend. We read in the destination there anyway.